src/AzureClient/Visualization/CloudJobEncoders.cs

```diff
@@ -51,9 +52,63 @@ internal static class CloudJobExtensions
                     ("Creation Time", cloudJob => cloudJob.Details.CreationTime?.ToString() ?? string.Empty),
                     ("Begin Execution Time", cloudJob => cloudJob.Details.BeginExecutionTime?.ToString() ?? string.Empty),
                     ("End Execution Time", cloudJob => cloudJob.Details.EndExecutionTime?.ToString() ?? string.Empty),
+                    ("Cost Estimate", cloudJob => cloudJob.GetCostEstimateText()),
                 },
                 Rows = jobsList.OrderByDescending(job => job.Details.CreationTime).ToList(),
             };
+
+        internal static string GetCostEstimateText(this CloudJob cloudJob) =>
+            cloudJob?.Details?.CostEstimate == null ? String.Empty :
+                                                      CurrencyHelper.FormatValue(cloudJob.Details.CostEstimate?.CurrencyCode,
+                                                                                 cloudJob.Details.CostEstimate?.EstimatedTotal);
+    }
+
+    internal static class CurrencyHelper
+    {
+        private static Dictionary<string, CultureInfo> currencyCodeToCultureInfo = new Dictionary<string, CultureInfo>();
+        static CurrencyHelper()
+        {
+            foreach (var cultureInfo in CultureInfo.GetCultures(CultureTypes.AllCultures)
+                                                   .Where((c) => !string.IsNullOrEmpty(c.Name) && !c.IsNeutralCulture))
+            {
+                try
+                {
+                    var regionInfo = new RegionInfo(cultureInfo.Name);
+                    currencyCodeToCultureInfo.TryAdd(regionInfo.ISOCurrencySymbol, cultureInfo);
+                }
+                catch {}
+            }
+        }
+
+        public static CultureInfo? GetCultureInfoForCurrencyCode(string currencyCode)
+        {
+            if (currencyCodeToCultureInfo.TryGetValue(currencyCode, out var cultureInfo))
+            {
+                return cultureInfo;
+            }
+
+            return null;
+        }
+
+        public static string FormatValue(string? currencyCode, float? value)
```

Comment:
Depending on the scale of the particular currency, `float` can be numerically unstable. For currency values, `Decimal` may be preferable, especially if making equality comparisons.

Reply:
The data model is a float, so I'll keep as such for now.